python/caffe/layers.py

```diff
@@ -0,0 +1,93 @@
+from collections import OrderedDict
+import re
+
+from .proto import caffe_pb2
+from google import protobuf
+
+def uncamel(s):
+    """Convert CamelCase to underscore_case."""
+
+    return re.sub('(?!^)([A-Z]+)', r'_\1', s).lower()
+
+def assign_proto(proto, name, val):
+    if isinstance(val, list):
+        getattr(proto, name).extend(val)
+    else:
```

Comment:
To generate weight filler & bias filler, I would add the following here 
elif isinstance(val, dict):
        new_proto = getattr(proto, name)
        for k, v in val.iteritems():
            assign_proto(new_proto, k, v)